_data-prepper/peer-forwarder.md

```diff
@@ -0,0 +1,186 @@
+| `aws_region` | A `String` represents the AWS Region to use `ACM`, `S3` or `AWS Cloud Map`. Required if `use_acm_certificate_for_ssl` is set to `true` or `ssl_certificate_file` and `ssl_key_file` is `AWS S3` path or if `discovery_mode` is set to `aws_cloud_map`. |
+
+#### Example config
+
+```yaml
+peer_forwarder:
+  ssl: true
+  ssl_certificate_file: "<cert-file-path>"
+  ssl_key_file: "<private-key-file-path>"
+```
+
+## Authentication
+
+This section describes optional authentication.
+
+`authentication`(optional) : A `Map` that enables mTLS. It can either be `mutual_tls` or `unauthenticated`. The default value is `unauthenticated`.
+
+```yaml
+peer_forwarder:
+  authentication:
+    mutual_tls:
+```
+
+## Metrics
+
+This section describes Peer Forwarder's metrics capabilities, including timer, counter, and gauge.
+
+Core Peer Forwarder introduces the following custom metrics and all the metrics are prefixed by `core.peerForwarder`.
+
+### Timer
+
+This section describes the timer metrics feature.
+
+- `requestForwardingLatency`: measures latency of forwarding requests by the Peer Forwarder client.
+- `requestProcessingLatency`: measures latency of processing requests by Peer Forwarder server.
+
+### Counter
+
+See the table below for counter metric options.
+
+| Value | Description |
+| ----- | ----------- |
+| `requests`| Measures total number of forwarded requests. |
+| `requestsFailed`| Measures total number of failed requests. Requests with HTTP response code other than `200`. |
+| `requestsSuccessful`|  Measures total number of successful requests. Requests with HTTP response code `200`. |
+| `requestsTooLarge`| Measures total number of requests that are too large to be written to Peer Forwarder buffer. Requests with HTTP response code `413`. |
+| `requestTimeouts`| Measures the total number of requests that timed out while writing content to Peer Forwarder buffer. Requests with HTTP response code `408`. |
+| `requestsUnprocessable`| Measures total number of requests that failed due to unprocessable entity. Requests with HTTP response code `422`. |
+| `badRequests`| Measures total number of requests with bad request format. Requests with HTTP response code `400`. |
+| `recordsSuccessfullyForwarded`| Measures total number of forwarded records successfully. |
+| `recordsFailedForwarding`| Measures total number of records failed to be forwarded. |
+| `recordsToBeForwarded` | Measures total number of records to be forwarded. |
+| `recordsToBeProcessedLocally` | Measures total number of records to be processed locally. |
+| `recordsActuallyProcessedLocally`| Measures total number of records actually processed locally. Sum of `recordsToBeProcessedLocally` and `recordsFailedForwarding`. |
+| `recordsReceivedFromPeers`| Measures total number of records received from remote peers. |
+
+### Gauge
+
+This section describes gauge metric options.
```

Comment:
Global: We don't need this type of introductory sentence in every section. It's sufficient to just provide the relevant text. Please revise where applicable.